### DISCUSSION TITLE
Differentiating the behaviour of single/double click on sprites on the Stage

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#6560] [Ideas]
In Scratch 1.4 there was a clear difference when you single- or double-clicked a sprite on the Stage:

- single click: triggered the "when this sprite clicked" hats (if any)
- double click: selected the sprite in the corral below the Stage

The second behaviour, that is just selection without triggering the "when this sprite clicked" hats, was very useful as you could select a sprite that you saw on the Stage in a project with many sprites (where finding the sprite in the corral is difficult) and you could analyze its scripts without making the project "go on".

Instead, in Scratch 2 and 3 the double click triggers both actions: the sprite is selected in the corral and the hats are triggered.

I propose differentiating the behaviour of single/double click on sprites on the Stage as it was in 1.4.

Moreover, it would be useful (for projects with many sprites) that when you double click a sprite on the Stage the corral window would scroll so to show the selected sprite.

**Comment by WorldLanguages** [comment]
> Instead, in Scratch 2 and 3 the double click triggers both actions: the sprite is selected in the corral and the hats are triggered.

This was probably decided because with the 1.4 behavior, all sprite clicks get added more latency as Scratch needed to wait to see if it was a double click or not 🤔 Is latency a problem here?

**Reply by DNin01** [reply]
We could minimize the lag by starting the double-click timeout as soon as the user released the mouse button and stopped it as soon as they clicked the second time.

**Comment by sfederici** [comment]
I don't know. But I wouldn't mind to wait a split second more when I click a sprite and this addon I proposed is enabled.

**Comment by DNin01** [comment]
Another alternative is Shift-clicking.

**Reply by sfederici** [reply]
Fine for me
